emacs/merlin.el

```diff
@@ -1662,21 +1662,37 @@ Empty string defaults to jumping to all these."
   "Return or update path of ocamlmerlin binary selected by configuration"
   (unless merlin-buffer-configuration
     (setq merlin-buffer-configuration (merlin--configuration)))
+
   (let ((command (merlin-lookup 'command merlin-buffer-configuration)))
     (unless command
-      (setq command (if (functionp merlin-command) (funcall merlin-command)
-                      merlin-command)))
-    (when (equal command 'opam)
-      (with-temp-buffer
-        (if (eq (call-process-shell-command
-                 "opam config var bin" nil (current-buffer) nil) 0)
-            (progn
-              (setq command (concat
-                             (replace-regexp-in-string "\n$" "" (buffer-string))
-                             "/ocamlmerlin"))
-              (push (cons 'command command) merlin-buffer-configuration))
-          (message "merlin-command: opam config failed (%S)" (buffer-string))
-          (setq command "ocamlmerlin"))))
+      (setq command
+            (cond ((functionp merlin-command) (funcall merlin-command))
+                  ((stringp merlin-command) merlin-command)
+                  ((equal merlin-command 'opam)
+                   (with-temp-buffer
+                     (if (eq (call-process-shell-command "opam config var bin" nil (current-buffer) nil) 0)
+                         (progn
+                           ;; the opam bin dir needs to be on the path, so if merlin
+                           ;; calls out to sub binaries (e.g. ocamlmerlin-reason), the
+                           ;; correct version is used rather than the version that
+                           ;; happens to be on the path
+
+                           ;; this was originally done via `opam exec' but that doesnt
+                           ;; work for opam 1, and added a performance hit
+                           (setq merlin-opam-bin-path (list (concat "PATH=" (string-trim (buffer-string)))))
```

Comment:
That's probably over 80c too, isn't it?

Reply:
Also, why is the `(replace-regexp-in-string "\n$" ...)` needed for setting `command`, but not for `merlin-opam-bin-path`?

Reply:
Good point. I've used that in both places now as of the last commit - originally I had a `string-trim` but perhaps there's a reason the `replace-regexp-in-string "\n$"` was used originally so I'll stick with that, thanks.